## Importing necessary libraries

In [ ]:
# Libraries to help with reading and manipulating data
import pandas as pd
import numpy as np

# Libraries to help with data visualization
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

# Library to help with statistical analysis
import scipy.stats as stats

## Sampling Distribution

**Q1. Suppose an automobile battery manufacturer claims that the mean lifetime of their battery is 60 months with a standard deviation of 6 months. Suppose the distribution of battery life is approximately normal. Find the probability that the mean lifetime of 40 randomly sampled batteries will be less than 58 months.**

The assumptions of the z-test are satisfied:


*   The population distribution is normal
*   The sample size (40) is greater than 30
*   The standard deviation (6) is known

In [ ]:
# import the required function
from scipy.stats import norm

# set the values of population mean and population standard deviation to 60 and 6 respectively
mu, sigma = 60, 6

# sample size = 40
n = 40

# set the value of the sample mean to 58
x_bar = 58

# calculate the test statistic
test_stat = (x_bar - mu) / (sigma/np.sqrt(n))

# find the probability that the mean lifetime of 40 randomly sampled batteries will be less than 58 months
prob_less_than_58 = norm.cdf(test_stat)
prob_less_than_58

0.017507490509831244

1.75% probability that the mean lifetime of 40 randomly samples batteries will be less than 58 months

## Interval Estimation

**Q2. A random sample of 40 households was selected as part of a study on electricity usage, and the number of kilowatt-hours (kWh) was recorded for each household in the sample for the first quarter of 2020. The average usage was found to be 310 kWh. In a very large study in the first quarter of the previous year, it was found that the standard deviation of the usage was 89 kWh.**

**Assuming the standard deviation is unchanged and that the usage is normally distributed, provide an expression for calculating a 95% confidence interval for the mean usage in the first quarter of 2019.**

In [ ]:
# import the required function
from scipy.stats import norm

# set the values of sample mean and sigma
x_bar, sigma = 310, 89

# set the value of sample size
n = 40

# construct the confidence interval
np.round(norm.interval(0.95, loc=x_bar, scale=sigma / np.sqrt(n)), 2)

array([282.42, 337.58])

The 95% confidence interval for the mean usage ranges from 282.42 to 337.58 kWh.

## Hypothesis Testing

**Q3. You are a manager of a Chinese restaurant. You want to determine whether the mean waiting time to place an order has changed in the past month from its previous population mean value of 4.5 minutes. State the null and alternative hypotheses.**

Null Hypothesis: Mean waiting time in the past month = 4.5 minutes

Alternative Hypothesis: Mean waiting time in the past month <> 4.5 minutes

**Q4. What is the p-value in a two-tailed z-test for one sample, where the computed test statistic (z-stat) is equal to +2.00?**

In [ ]:
# import the required function
from scipy.stats import norm

# calculate the p-value using the cdf() function
2*(1 - norm.cdf(2))

0.04550026389635842

The p-value in a two-tailed z-test for one sample, where the z-stat is +2.00 is 4.55%

**Q5. Samy, Product Manager of K2 Jeans, wants to launch a product line into a new market area. A Survey of a random sample of 400 households in that market showed a mean income per household of 30000 rupees. The standard deviation based on an earlier pilot study of households is 8000 rupees. Samy strongly believes the product line will be adequately profitable only in markets where the mean household income is greater than 29000 rupees. Samy wants our help in deciding whether the product line should be introduced in the new market. Perform statistical analysis and based on that draw a conclusion.**

Assume a level of significance ($\alpha$) of 5%

In [ ]:
# set the values of population mean and standard deviation to 29000 and 8000 respectively
mu, sigma = 29000, 8000

# set the value sample mean to 30000
x_bar = 30000

# set sample size to 400
n = 400

# calculate the test statistic
test_stat = (x_bar - mu) / (sigma/np.sqrt(n))

print("The test statistic is ", test_stat)

#calculate the critical value
critical_val = norm.ppf(0.95)
print("The critical value is ", critical_val)

The test statistic is  2.5
The critical value is  1.6448536269514722


Since test statistic is greater than the critical value and, thus, lies in the reject zone, the null hypothesis can be rejected. Therefore, Samy may go ahead and introduce the product line since the mean household will most likely be greater than 29000 rupees.